src/core/agent/agent.ts

```diff
@@ -316,8 +320,8 @@ class Agent {
 
     this.signifyClient = new SignifyClient(connectUrl, bran, Tier.low);
     this.agentServicesProps.signifyClient = this.signifyClient;
-    
-    await this.signifyClient.connect().catch(error => {
+
+    await this.signifyClient.connect().catch((error) => {
```

Comment:
![Screenshot 2024-10-16 at 09 35 43](https://github.com/user-attachments/assets/8bd81191-da86-4455-94eb-6d507e76b15e)

We have duplicate code here x3, `start`, `bootAndConnect`, `recoverKeriaAgent`. Maybe we can refactor it like:

```js
private async connectSignifyClient(): Promise<void> {
    try {
      await this.signifyClient.connect();
      this.markAgentStatus(true);
    } catch (error) {
      console.error(error);
      if (error.message === "Failed to fetch") {
        throw new Error(Agent.KERIA_CONNECT_FAILED_BAD_NETWORK);
      }
      const status = error.message.split(" - ")[1];
      if (/404/gi.test(status)) {
        throw new Error(Agent.KERIA_NOT_BOOTED);
      }
      throw new Error(Agent.KERIA_BOOTED_ALREADY_BUT_CANNOT_CONNECT);
    }
  }
```
And the call it like:

```js
async start(keriaConnectUrl: string): Promise<void> {
    if (!Agent.isOnline) {
      await signifyReady();
      const bran = await this.getBran();
      this.signifyClient = new SignifyClient(keriaConnectUrl, bran, Tier.low);
      this.agentServicesProps.signifyClient = this.signifyClient;
      await this.connectSignifyClient();
    }
  }

```

Reply:
good point! updated!

I just did it for the connect, and kept `markAgentStatus` out of it, as I only want to mark the status as online once everything is done, and in some cases we want to store in key store etc after connecting